### PR TITLE
Fixing return of incorrect subprocess

### PR DESCRIPTION
**Pull request by roigcarlo**
This fixes the run tests script not returning any error code while failing to execute a subprocess due to the command being invalid. Additionally I added to return an error if the subprocess can be initiated but the command fails due to wrong number of arguments (which should never happen)

I also fixed a warning with the imp lib being deprecated. This should increase compatibility with python 2.x and 3.3 or lower.

Finally, now all executions timed or not, go through the same pipeline to prevent future errors like this.

### CODE REVIEW

**Comment by philbucher** [comment]
Nice work!
Are you working on more updates of the testing? I am asking bcs I am also currently trying to better integrate the c-tests (remember the issue we have).
Just asking to prevent merging struggles

**Comment by roigcarlo** [comment]
Hi @philbucher, we plan to integrate support for mpi, but honestly I don't see it done before christmas

**Comment by philbucher** [comment]
ok thx

File: kratos/python_scripts/run_tests.py [hunk 1]
```diff
@@ -42,7 +40,7 @@ def GetModulePath(module):
 
     '''
 
-    return imp.find_module(module)[1]
+    return os.path.dirname(KtsMp.__file__)
```
Comment (philbucher) [inline kratos/python_scripts/run_tests.py]: This is no longer general I'd say => `GetModulePath` should now be named `GetKratosMultiphysicsPath`

**Comment by pooyan-dadvand** [comment]
Just to say that it works for me and now reports correctly an error in subprocess

**Comment by philbucher** [comment]
will this be merged this year?
Asking bcs I have some developments touching the same files

**Comment by philbucher** [comment]
@roigcarlo I used the new functions from #3738 

Also I noticed that if there is an error in one of the python-tests, running of all tests is aborted! (unrelated to what I changed)
E.g. if there is an error in the core it doesn't run the tests in the apps
or if there is an error in an app it won't run the tests in the remaining apps

**Comment by roigcarlo** [comment]
@philbucher Agree with the changes.

Also, the premature exit is caused in one of the changes I introduced in this PR, but honestly I don't remember if it was our intention or a bug, because its even stated that the program should exit. But I find it odd too

This is the change:
```python
if self.exitCode != 0:
    # If thread terminated with an error, propagate and terminate
    sys.exit(self.exitCode)
```

Let me speak with @pooyan-dadvand before reverting it

**Comment by roigcarlo** [comment]
Hi @philbucher, I spoke with @pooyan-dadvand and he agreed to let the other tests continue, I will change it this afternoon

**Comment by philbucher** [comment]
ping

**Comment by philbucher** [comment]
ping :)

**Comment by pooyan-dadvand** [comment]
I assume that the ping is for @roigcarlo

**Comment by philbucher** [comment]
> I assume that the ping is for @roigcarlo

indeed it is :)

**Comment by roigcarlo** [comment]
Give me one sec

**Comment by roigcarlo** [comment]
Done

**Comment by philbucher** [comment]
Thanks @roigcarlo I am testing it now

**Review by philbucher — APPROVED** [review]
works again, thx!

**Pull request merged**